autotest/gcore/testnonboundtoswig.py

```diff
@@ -32,14 +32,10 @@
 from osgeo import gdal
 import pytest
 
-
-try:
-    import ctypes
-except ImportError:
-    pass
-
 import gdaltest
 
+ctypes = pytest.importorskip('ctypes')
+
 gdal_handle_init = False
 gdal_handle = None
 gdal_handle_stdcall = None
```

Comment:
It's really unclear why we're loading both cdll + windll on windows, then randomly alternating calls between them? 

I wonder if it's better to restructure this as a fixture that returns a different handle depending on OS, then use that in the test methods below.

Reply:
Could be a separate PR though